diff --git a/python/stubs-out/mrpt/pymrpt/__init__.pyi b/python/stubs-out/mrpt/pymrpt/__init__.pyi
index 7f7caeb4c..5edcfd064 100644
--- a/python/stubs-out/mrpt/pymrpt/__init__.pyi
+++ b/python/stubs-out/mrpt/pymrpt/__init__.pyi
@@ -1,5 +1,10 @@
+from . import mrpt
+from . import std
+
+
 class GLFWwindow:
     def __init__(self) -> None: ...
 
+
 class NVGcontext:
     def __init__(self) -> None: ...
diff --git a/python/stubs-out/mrpt/pymrpt/mrpt/__init__.pyi b/python/stubs-out/mrpt/pymrpt/mrpt/__init__.pyi
index eec5ec781..fad622fdb 100644
--- a/python/stubs-out/mrpt/pymrpt/mrpt/__init__.pyi
+++ b/python/stubs-out/mrpt/pymrpt/mrpt/__init__.pyi
@@ -1,3 +1,9 @@
+from . import comms, io, obs, opengl, poses, rtti, system, typemeta
+from . import apps, bayes, config, containers, cpu, expr, global_settings
+from . import graphs, gui, hwdrivers, img, kinematics, maps, math, nav
+from . import random, serialization, slam, tfest, topography, vision
+
+
 from typing import Any, ClassVar, List
 
 from typing import overload
