diff --git a/python/src/std/array.cpp b/python/src/std/array.cpp
index 680fd31c3..173995ed6 100644
--- a/python/src/std/array.cpp
+++ b/python/src/std/array.cpp
@@ -28,6 +28,8 @@
 	PYBIND11_MAKE_OPAQUE(std::shared_ptr<void>)
 #endif
 
+PYBIND11_MAKE_OPAQUE(std::array<mrpt::img::CImage,6UL>)
+
 void bind_std_array(std::function< pybind11::module &(std::string const &namespace_) > &M)
 {
 	{ // std::array file:array line:95
diff --git a/python/src/std/stl_deque.cpp b/python/src/std/stl_deque.cpp
index cabe86ec7..302018560 100644
--- a/python/src/std/stl_deque.cpp
+++ b/python/src/std/stl_deque.cpp
@@ -18,6 +18,9 @@
 	PYBIND11_MAKE_OPAQUE(std::shared_ptr<void>)
 #endif
 
+PYBIND11_MAKE_OPAQUE(std::deque<mrpt::bayes::CProbabilityParticle<mrpt::math::TPose2D, mrpt::bayes::particle_storage_mode::VALUE>>)
+PYBIND11_MAKE_OPAQUE(std::deque<mrpt::bayes::CProbabilityParticle<mrpt::math::TPose3D, mrpt::bayes::particle_storage_mode::VALUE>>)
+
 void bind_std_stl_deque(std::function< pybind11::module &(std::string const &namespace_) > &M)
 {
 	{ // std::deque file:bits/stl_deque.h line:767
diff --git a/python/src/std/stl_deque_1.cpp b/python/src/std/stl_deque_1.cpp
index d974577d3..a80326114 100644
--- a/python/src/std/stl_deque_1.cpp
+++ b/python/src/std/stl_deque_1.cpp
@@ -26,6 +26,9 @@
 	PYBIND11_MAKE_OPAQUE(std::shared_ptr<void>)
 #endif
 
+PYBIND11_MAKE_OPAQUE(std::deque<mrpt::math::TPose3D>)
+PYBIND11_MAKE_OPAQUE(std::deque<mrpt::bayes::CProbabilityParticle<mrpt::maps::CRBPFParticleData, mrpt::bayes::particle_storage_mode::POINTER>>)
+
 void bind_std_stl_deque_1(std::function< pybind11::module &(std::string const &namespace_) > &M)
 {
 	{ // std::deque file:bits/stl_deque.h line:767
diff --git a/python/src/std/stl_deque_2.cpp b/python/src/std/stl_deque_2.cpp
index ca96e381d..54e7eaaf4 100644
--- a/python/src/std/stl_deque_2.cpp
+++ b/python/src/std/stl_deque_2.cpp
@@ -16,6 +16,8 @@
 	PYBIND11_MAKE_OPAQUE(std::shared_ptr<void>)
 #endif
 
+PYBIND11_MAKE_OPAQUE(std::deque<mrpt::system::CDirectoryExplorer::TFileInfo>)
+
 void bind_std_stl_deque_2(std::function< pybind11::module &(std::string const &namespace_) > &M)
 {
 	{ // std::deque file:bits/stl_deque.h line:767
diff --git a/python/src/std/stl_map.cpp b/python/src/std/stl_map.cpp
index db3f49fde..793bf72f3 100644
--- a/python/src/std/stl_map.cpp
+++ b/python/src/std/stl_map.cpp
@@ -34,6 +34,11 @@
 	PYBIND11_MAKE_OPAQUE(std::shared_ptr<void>)
 #endif
 
+PYBIND11_MAKE_OPAQUE(std::map<std::string,std::string>)
+PYBIND11_MAKE_OPAQUE(std::map<std::string,double>)
+PYBIND11_MAKE_OPAQUE(std::map<double,double>)
+PYBIND11_MAKE_OPAQUE(std::map<std::string,mrpt::poses::CPose3D>)
+
 void bind_std_stl_map(std::function< pybind11::module &(std::string const &namespace_) > &M)
 {
 	{ // std::map file:bits/stl_map.h line:100
diff --git a/python/src/std/stl_multimap.cpp b/python/src/std/stl_multimap.cpp
index ddbd20205..ccfa54684 100644
--- a/python/src/std/stl_multimap.cpp
+++ b/python/src/std/stl_multimap.cpp
@@ -28,6 +28,10 @@
 	PYBIND11_MAKE_OPAQUE(std::shared_ptr<void>)
 #endif
 
+PYBIND11_MAKE_OPAQUE(std::multimap<mrpt::Clock::time_point, mrpt::serialization::CSerializable::Ptr>)
+PYBIND11_MAKE_OPAQUE(std::map<mrpt::containers::yaml::node_t,mrpt::containers::yaml::node_t>)
+
+
 void bind_std_stl_multimap(std::function< pybind11::module &(std::string const &namespace_) > &M)
 {
 	{ // std::multimap file:bits/stl_multimap.h line:99
diff --git a/python/src/std/stl_vector.cpp b/python/src/std/stl_vector.cpp
index fe00b811e..e9110c758 100644
--- a/python/src/std/stl_vector.cpp
+++ b/python/src/std/stl_vector.cpp
@@ -17,6 +17,8 @@
 	PYBIND11_MAKE_OPAQUE(std::shared_ptr<void>)
 #endif
 
+PYBIND11_MAKE_OPAQUE(std::vector<std::string>)
+
 void bind_std_stl_vector(std::function< pybind11::module &(std::string const &namespace_) > &M)
 {
 	{ // std::vector file:bits/stl_vector.h line:389
