--- generated-sources-pybind/mrpt/obs/gnss_messages_type_list.cpp	2023-04-27 22:41:46.499949717 +0200
+++ generated-sources-pybind/mrpt/obs/gnss_messages_type_list.new.cpp	2023-04-27 22:41:37.404072257 +0200
@@ -21,6 +21,9 @@
 #include <string>
 #include <pybind11/stl.h>
 
+#include <mrpt/obs/gnss_messages.h>
+
+
 #ifndef BINDER_PYBIND11_TYPE_CASTER
 	#define BINDER_PYBIND11_TYPE_CASTER
 	PYBIND11_DECLARE_HOLDER_TYPE(T, std::shared_ptr<T>)
