diff --git a/python/stubs-out/mrpt/pymrpt/mrpt/opengl/__init__.pyi b/python/stubs-out/mrpt/pymrpt/mrpt/opengl/__init__.pyi
index 8abd014c7..d29cf054a 100644
--- a/python/stubs-out/mrpt/pymrpt/mrpt/opengl/__init__.pyi
+++ b/python/stubs-out/mrpt/pymrpt/mrpt/opengl/__init__.pyi
@@ -1,5 +1,7 @@
 from typing import Any, ClassVar, Dict, List, Optional, Tuple
 
+from . import stock_objects
+
 from typing import overload
 import mrpt.pymrpt.mrpt.containers
 import mrpt.pymrpt.mrpt.img
diff --git a/python/stubs-out/mrpt/pymrpt/mrpt/poses/__init__.pyi b/python/stubs-out/mrpt/pymrpt/mrpt/poses/__init__.pyi
index 09ba314ee..6d716eb80 100644
--- a/python/stubs-out/mrpt/pymrpt/mrpt/poses/__init__.pyi
+++ b/python/stubs-out/mrpt/pymrpt/mrpt/poses/__init__.pyi
@@ -1,5 +1,7 @@
 from typing import Any, ClassVar, Dict, Tuple
 
+from . import Lie
+
 from typing import overload
 import mrpt.pymrpt.mrpt
 import mrpt.pymrpt.mrpt.bayes
